cryptoservice/crypto_service.go

```diff
@@ -110,28 +100,23 @@ func (ccs *CryptoService) Sign(keyIDs []string, payload []byte) ([]data.Signatur
 			}
 		}
 
-		algorithm := privKey.Algorithm()
 		var sigAlgorithm data.SigAlgorithm
-		var sig []byte
 
-		switch algorithm {
-		case data.RSAKey:
-			sig, err = rsaSign(privKey, payload)
+		switch privKey.(type) {
+		case *data.RSAPrivateKey, *data.RSAx509PrivateKey:
```

Comment:
I made everything very explicit so the RSAx509PrivateKey and ECDSAx509PrivateKey exist specifically as private counterparts to the x509 variants of the public keys. It's possible we don't need them, I'm not 100% certain though. I'll continue to assess but it's something that can safely be removed at a later time if we confirm they're not required.